Source/Scene/Primitive.js

```diff
@@ -1119,57 +1121,79 @@ define([
                 });
             }
 
+            var transferTerrainHeightsPromise;
             if (!defined(createGeometryTaskProcessors)) {
                 createGeometryTaskProcessors = new Array(numberOfCreationWorkers);
                 for (i = 0; i < numberOfCreationWorkers; i++) {
                     createGeometryTaskProcessors[i] = new TaskProcessor('createGeometry', Number.POSITIVE_INFINITY);
                 }
+                // Push tasks transferring the JSON to the workers.
+                transferTerrainHeightsPromise = ApproximateTerrainHeights.initialize()
+                    .then(function() {
+                        var terrainHeightsString = JSON.stringify(ApproximateTerrainHeights._terrainHeights);
+                        var workerTerrainHeightsPromises = new Array(numberOfCreationWorkers);
+                        for (i = 0; i < numberOfCreationWorkers; i++) {
+                            workerTerrainHeightsPromises[i] = createGeometryTaskProcessors[i].scheduleTask({
+                                subTasks : [
+                                    {
+                                        moduleName : 'parseTerrainHeights',
+                                        geometry : terrainHeightsString
+                                    }
+                                ]
+                            });
+                        }
+                        return when.all(workerTerrainHeightsPromises);
+                    });
+            } else {
+                transferTerrainHeightsPromise = when();
             }
 
-            var subTask;
-            subTasks = subdivideArray(subTasks, numberOfCreationWorkers);
-
-            for (i = 0; i < subTasks.length; i++) {
-                var packedLength = 0;
-                var workerSubTasks = subTasks[i];
-                var workerSubTasksLength = workerSubTasks.length;
-                for (j = 0; j < workerSubTasksLength; ++j) {
-                    subTask = workerSubTasks[j];
-                    geometry = subTask.geometry;
-                    if (defined(geometry.constructor.pack)) {
-                        subTask.offset = packedLength;
-                        packedLength += defaultValue(geometry.constructor.packedLength, geometry.packedLength);
-                    }
-                }
-
-                var subTaskTransferableObjects;
-
-                if (packedLength > 0) {
-                    var array = new Float64Array(packedLength);
-                    subTaskTransferableObjects = [array.buffer];
+            transferTerrainHeightsPromise.then(function() {
```

Comment:
Note to reviewer: diff tool needs to chill here.

Reply:
Adding the query param `w=1` will make the diff ignore whitespace changes
https://github.com/AnalyticalGraphicsInc/cesium/pull/6716/files?w=1